commit e37988bcd1fddb726cf08e4f04a1b8dc4a2f80aa
Author: Justin Stitt <justinstitt@google.com>
Date:   Mon Oct 23 19:14:49 2023 +0000

    s390/sclp: replace deprecated strncpy with strtomem
    
    Let's move away from using strncpy() as it is deprecated [1].
    
    Instead use strtomem() as `e.id` is already marked as nonstring:
    |       char id[4] __nonstring;
    
    We don't need strtomem_pad() because `e` is already memset to 0 --
    rendering any additional NUL-padding useless.
    
    Link: https://www.kernel.org/doc/html/latest/process/deprecated.html#strncpy-on-nul-terminated-strings [1]
    Link: https://github.com/KSPP/linux/issues/90
    Cc: linux-hardening@vger.kernel.org
    Signed-off-by: Justin Stitt <justinstitt@google.com>
    Reviewed-by: Kees Cook <keescook@chromium.org>
    Link: https://lore.kernel.org/r/20231023-strncpy-drivers-s390-char-sclp-c-v1-1-eaeef80522bb@google.com
    Signed-off-by: Vasily Gorbik <gor@linux.ibm.com>

diff --git a/drivers/s390/char/sclp.c b/drivers/s390/char/sclp.c
index ba9b202c5dee..d53ee34d398f 100644
--- a/drivers/s390/char/sclp.c
+++ b/drivers/s390/char/sclp.c
@@ -81,7 +81,7 @@ static inline void sclp_trace(int prio, char *id, u32 a, u64 b, bool err)
 	struct sclp_trace_entry e;
 
 	memset(&e, 0, sizeof(e));
-	strncpy(e.id, id, sizeof(e.id));
+	strtomem(e.id, id);
 	e.a = a;
 	e.b = b;
 	debug_event(&sclp_debug, prio, &e, sizeof(e));